salt/minion.py

```diff
@@ -585,9 +585,10 @@ def __init__(self, opts):
         # Clean out the proc directory (default /var/cache/salt/minion/proc)
         if (self.opts.get('file_client', 'remote') == 'remote'
                 or self.opts.get('use_master_when_local', False)):
-            if HAS_ZMQ:
-                zmq.eventloop.ioloop.install()
-            io_loop = LOOP_CLASS.current()
+            if self.opts['transport'] == 'zeromq' and HAS_ZMQ:
+                io_loop = zmq.eventloop.ioloop.ZMQIOLoop()
+            else:
+                io_loop = LOOP_CLASS.current()
```

Comment:
I know this is already merged. I just wanted to point out that the 4 lines above are functionally equivalent to simply:

```
io_loop = LOOP_CLASS.current()
```

The reason is that when `HAS_ZMQ` is true, `LOOP_CLASS` is: `zmq.eventloop.ioloop.ZMQIOLoop`. So even if `self.opts['transport'] != 'zeromq'`, it will use the `else` case which is functionally the same since `LOOP_CLASS` is `zmq.eventloop.ioloop.ZMQIOLoop`. The only real difference in the code from this change and the original code is that you no longer call `zmq.eventloop.ioloop.install()` in the `HAS_ZMQ` case. So by simply removing the lines:

```
if HAS_ZMQ:
    zmq.eventloop.ioloop.install()
```

You would have accomplished the same result, but in a more straight forward way.